### ISSUE TITLE
Named window getter is missing

### ISSUE THREAD

**Issue by jdm** [servo/servo#27949]
From https://html.spec.whatwg.org/multipage/window-object.html#the-window-object, we are currently missing this bit:
```
  getter object (DOMString name);
  // Since this is the global object, the IDL named getter adds a NamedPropertiesObject exotic
  // object on the prototype chain. Indeed, this does not make the global object an exotic object.
  // Indexed access is taken care of by the WindowProxy exotic object.
```
and `LegacyUnenumerableNamedProperties`

Spec:
* https://html.spec.whatwg.org/multipage/window-object.html#named-access-on-the-window-object
* https://heycam.github.io/webidl/#LegacyUnenumerableNamedProperties

https://github.com/servo/servo/pull/25562 and https://github.com/servo/servo/pull/21869 are two previous WIPs to implement this feature.

**Comment by jdm** [comment]
The most recent work on this was happening in https://github.com/servo/servo/tree/named-getter.

**Comment by pylbrecht** [comment]
Sign me up, please!

**Comment by pylbrecht** [comment]
Alright, I'm gonna dive into the provided information and probably come back with some questions.

**Comment by pshaughn** [comment]
I think I had come very close in #25562, although my implementation was inelegant. The only thing I had outright wrong in it was the shape of the Window object's prototype chain, which requires some subtle Spidermonkey knowledge I didn't have and still don't have.

**Comment by pylbrecht** [comment]
I ended up in the terrain of debugging code I did not write nor fully understand, which is quite frustrating and took out all the fun for me. I'd like to hand this over to someone else with more experience or more will to debug this.

Besides https://servo.zulipchat.com/#narrow/stream/263398-general/topic/Debugging.20wpt.20tests hopefully giving some context on what I've tried so far, I'm of course available to answer any questions to anyone picking this up.

@jdm, I'm not sure what to do about https://github.com/servo/servo/pull/27952. Feel free to close it, if that is the way to go.
